bench/wtperf/wtperf_truncate.c

```diff
@@ -145,11 +149,28 @@ run_truncate(CONFIG *cfg, CONFIG_THREAD *thread,
 	trunc_cfg->last_total_inserts = trunc_cfg->total_inserts;
 
 	/* We are done if there isn't enough data to trigger a new milestone. */
-	if (trunc_cfg->expected_total <= trunc_cfg->needed_stones)
+	if (trunc_cfg->expected_total <= thread->workload->truncate_count)
 		return (0);
 
+	used_stone_gap = trunc_cfg->stone_gap;
```

Comment:
I'd find this code easier to follow if it were shuffled a bit. So that ``used_stone_gap`` is calculated once after the multiplier has been adjusted. i.e move this line below the following if clause, and make it:

```
used_stone_gap =  trunc_cfg->stone_gap * trunc_cfg->catchup_multiplier;
```